lib/simple-connect-policy.js

```diff
@@ -18,14 +21,25 @@ module.exports = class SimpleConnectPolicy {
   //         'name@version' or { name, version }, where version
   //         must be a valid semver range
   //   connection - JSTP connection
+  //   session - Session object to reconnect to existing session (optional)
   //   callback - callback function that has signature
   //              (error, connection)
   //
-  connect(app, connection, callback) {
-    connection.handshake(
-      app, this.login, this.password,
-      (error) => {
-        callback(error, connection);
-      });
+  connect(app, connection, session, callback) {
+    if (!session) {
```

Comment:
This if can be removed completely by assigning this.login to session parameter if `!session` and using handshake method with session and this.password and 2, 3 paraments. Also currently callback in a first if branch passes invalid session object (should be this.login or null).